File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports WindowsApplication1.CustomEditor

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
		End Function

		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = CreateTable(20)
			ButtonEditCustomCaptionHelper.Register(gridView1.Columns(0), AddressOf OnGetButtonCaption)
			ButtonEditCustomCaptionHelper.Register(gridView1.Columns(1), AddressOf OnGetButtonCaption)
		End Sub

		Public Sub OnGetButtonCaption(ByVal helper As ButtonEditCustomCaptionHelper, ByVal args As GetButtonCaptionEventArgs)
			If args.RowHandle = 2 Then
				Return
			End If
			args.ButtonText = String.Format("{0};{1};Button{2}", args.Column.VisibleIndex, args.RowHandle, args.Button.Index)
		End Sub
	End Class
End Namespace

File: CS/CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using WindowsApplication1.CustomEditor;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }

        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
            ButtonEditCustomCaptionHelper.Register(gridView1.Columns[0], OnGetButtonCaption);
            ButtonEditCustomCaptionHelper.Register(gridView1.Columns[1], OnGetButtonCaption);
        }

        public void OnGetButtonCaption(ButtonEditCustomCaptionHelper helper, GetButtonCaptionEventArgs args)
        {
            if (args.RowHandle == 2)
                return;
            args.ButtonText = String.Format("{0};{1};Button{2}", args.Column.VisibleIndex, args.RowHandle, args.Button.Index);
        }
    }
}